compare-oracle-to-fallback/src/agent.spec.js

```diff
@@ -1,53 +1,52 @@
+  },
+}));
+
 // required libraries
+const {
+  BlockEvent, FindingType, FindingSeverity, Finding,
+} = require('forta-agent');
 const BigNumber = require('bignumber.js');
-const { createBlockEvent } = require('forta-agent');
-
-// mock the initializeTokensContractsAlerts async function before importing the agent module
-// the agent module will then receive our mocked version of the function when it is imported
-jest.mock('./agent-setup', () => ({
-  ...jest.requireActual('./agent-setup'),
-  initializeTokensContractsAlerts: jest.fn().mockResolvedValue(),
-}));
 
-const { createAlert, calculatePercentError } = require('./agent-setup');
-const { provideHandleBlock } = require('./agent');
+const { provideInitialize, provideHandleBlock, calculatePercentError } = require('./agent');
+const config = require('../bot-config.json');
 
-const { compareOracleToFallback } = require('../agent-config.json');
+const ALERT_MINIMUM_INTERVAL_SECONDS = config.alertMinimumIntervalSeconds;
 
-const ALERT_MINIMUM_INTERVAL_SECONDS = compareOracleToFallback.alertMinimumIntervalSeconds;
+function createBlockEvent(block) {
+  return new BlockEvent(0, 1, block);
+}
 
 describe('Aave oracle versus fallback oracle agent', () => {
+  let initializeData;
   let handleBlock;
 
-  function mockTokensContractsAlertsPromise(assetPriceOracle, assetPriceFallback, tStart) {
-    // create a token
-    const reserveToken = { symbol: 'FAKE1', tokenAddress: '0xFIRSTFAKETOKENADDRESS' };
-
-    // create a mock price oracle contract for returning the value we want to test with
-    const aaveOracleContractInstance = {
-      getAssetPrice: jest.fn(() => Promise.resolve(assetPriceOracle)),
-    };
-
-    // create a mock fallback oracle contract for returning the value we want to test with
-    const fallbackOracleContractInstance = {
-      getAssetPrice: jest.fn(() => Promise.resolve(assetPriceFallback)),
-    };
+  beforeEach(async () => {
+    initializeData = {};
 
-    // create an object to track the number of token alerts in the last day
-    const tokenAlert = { numAlertsInLastDay: 0, tStart };
-
-    // create the Array of Tuples of tokens / oracle / fallback / alerts objects
-    const tokenContractFallbackAlertTuples = [
-      {
-        reserveToken, aaveOracleContractInstance, fallbackOracleContractInstance, tokenAlert,
-      },
-    ];
-
-    // wrap the Array of Tuples in a Promise (which we will resolve immediately)
-    // the original function returns a Promise that is resolved when all contract interactions
-    // have finished
-    return Promise.resolve(tokenContractFallbackAlertTuples);
-  }
+    // initialize the handler
+    await (provideInitialize(initializeData))();
+    handleBlock = provideHandleBlock(initializeData);
+  });
 
   describe('Fallback Price Oracle Monitoring', () => {
     it('returns findings if percent error between oracle and fallback is > 2% and last alert was created more than 24 hours ago', async () => {
```

Comment:
Sorry this wasn't done in the original implementation (I think this is my code), but manually overriding the threshold that is loaded from the configuration file is probably the correct approach.

In that case, the string in the `it()` statement should be changed to something like:
```suggestion
    it('returns findings if percent error between oracle and fallback is greater than the threshold and the last alert was created more than 24 hours ago', async () => {
```